File: frontend/src/types/sync-schedule.ts
export type SyncJobType =
  | "jellyfin_users_import"
  | "radarr_import"
  | "jellyfin_import_movies"
  | "jellyfin_movie_watch_history"
  | "sonarr_import"
  | "jellyfin_import_series"
  | "jellyfin_series_watch_history";

export type SchedulePreset = "daily" | "weekly" | "monthly" | "custom";

export interface SyncScheduleResponse {
  job_type: SyncJobType;
  preset: SchedulePreset;
  cron_expression: string;
  is_enabled: boolean;
  is_running: boolean;
  last_run_at: string | null;
  next_run_at: string | null;
}

export interface SyncScheduleListResponse {
  schedules: SyncScheduleResponse[];
}

export interface SyncScheduleRequest {
  preset: SchedulePreset;
  cron_expression?: string;
}

File: backend/app/schemas/sync_schedule.py
"""Pydantic schemas for sync schedule endpoints."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models import SchedulePreset, SyncJobType


class SyncScheduleRequest(BaseModel):
    preset: SchedulePreset
    cron_expression: str | None = Field(default=None, max_length=100)


class SyncScheduleResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    job_type: SyncJobType
    preset: SchedulePreset
    cron_expression: str
    is_enabled: bool
    is_running: bool
    last_run_at: datetime | None
    next_run_at: datetime | None


class SyncScheduleListResponse(BaseModel):
    schedules: list[SyncScheduleResponse]


class SyncTriggerResponse(BaseModel):
    job_type: SyncJobType
    message: str
